snuba/query/processors/discover_column_processor.py

```diff
@@ -0,0 +1,91 @@
+from typing import Callable, Mapping, Optional
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.query.dsl import multiply
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+EVENTS = "events"
+TRANSACTIONS = "transactions"
+
+
+class DiscoverColumnProcessor(QueryProcessor):
```

Comment:
Thank you for doing this, but, unfortunately, this cannot work as a dataset QueryProcessor.
In the query processing pipeline, storage selection happens after the dataset processors have been executed, so you cannot make the logic of your query processor depend on such selection.
You cannot simply add this to the two storages either. 
The problem is that this logic is supposed to be in a component that does not exist: the query translator, which is supposed to operate in the StorageQueryPlanner between storage selection and storage query processors.

So, if you want to still move this ahead before that component is built, I suggest you to split this query processor into two (one for events and one for transactions), subclass SelectedStorageQueryPlanBuilder and prepend the appropriate processor to the list of processors that will be applied to the storage in the build_plan method.

Otherwise you can wait for the proper abstraction to be built (probably matter of weeks) and resurrect this at that point.

Reply:
Hmmm seems like it might be smarter to wait then. Thanks! I'll hedge this.